src/core/SampleBuffer.cpp

```diff
@@ -957,13 +957,27 @@ void SampleBuffer::visualize( QPainter & _p, const QRect & _dr,
 	const int xb = _dr.x();
 	const int first = focus_on_range ? _from_frame : 0;
 	const int last = focus_on_range ? _to_frame : m_frames;
-	for( int frame = first; frame < last; frame += fpp )
+	if( m_reversed )
+	{
+		for( int frame = last; frame > first; frame -= fpp )
+		{
+			l[n] = QPointF( xb + ( (frame - first) * double( w ) / nb_frames ),
+				( yb - ( m_data[frame][0] * y_space * m_amplification ) ) );
+			r[n] = QPointF( xb + ( (frame - first) * double( w ) / nb_frames ),
+				( yb - ( m_data[frame][1] * y_space * m_amplification ) ) );
+			++n;
+		}
+	}
+	else
```

Comment:
Since the drawing statements are equal on both `for` loops, I wanted to know what you and others think of actually just manipulating the for statement, something like:

```
for (int frame = m_reversed ? last : first;
    m_reversed ? frame > first : frame < last;
    m_reversed ? frame -= fpp : frame += fpp)
```

I'm not sure if the cost on readability is really worth it for saving around 8 lines though, that is if the ternary operator can be used that way.

Reply:
Something I'm also struggling to get used to, but according to our code style guideline it should be `for(statements)` and the expressions inside `QPointF` shouldn't have spaces after `(` and before `)`

Reply:
I'll test if that works, and edit this comment with the result.

Reply:
Awesome! Well, as I said I'm not sure what others think of this approach in terms of readability, but if they agree and you're willing to change the other parts of the code that have similar loops sure! It would save work from a possible refactor in the future :smiley:

Reply:
Sure, I'll have to look into those snippets to see if that's possible, and if it is, I'll get on it tomorrow!

Reply:
Sorry for the massive delay, but I've now changed the three snippets. I used @PhysSong's idea that he proposed on Discord for my snippet, and the other two were just assigning a variable based on a ternary operator.